src/lib.rs

```diff
@@ -91,4 +91,48 @@ impl StrRange {
     pub fn end(self) -> StrIndex {
         self.end
     }
+
+    /// Do these ranges overlap?
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use str_index::*;
+    /// let left = StrRange::between(0.into(), 20.into());
+    /// let right = StrRange::between(10.into(), 30.into());
+    /// assert!(left.overlaps(right));
+    /// assert!(right.overlaps(left));
+    ///
+    /// let left = StrRange::between(0.into(), 10.into());
+    /// let right = StrRange::between(10.into(), 20.into());
+    /// assert!(!left.overlaps(right));
+    /// assert!(!right.overlaps(left));
+    ///
+    /// let outer = StrRange::between(0.into(), 20.into());
+    /// let inner = StrRange::between(5.into(), 15.into());
+    /// assert!(outer.overlaps(inner));
+    /// assert!(inner.overlaps(outer));
+    /// ```
+    pub fn overlaps(self, other: StrRange) -> bool {
```

Comment:
I think a shorter way to write overlaps is “not disjoint», where disjoint is end1 < start2 or end2 < start1.

Also the semantics here are a bit iffy: do ranges with unit intersection overlap?